Libmesh submodule update, add pymongo to tools, add --fast to mfem build

roystgnr opened the pull request:
LibMesh update summary:

```
    - Complete overhaul of `VariationalMeshSmoother` code.  This gives
      support for more finite element types and converges to more optimal
      meshes.
    - Enabled FE caching as appropriate for variable types with a
      dependency on edge/face orientation.  Added command-line
      `--disable-caching` option to help with debugging.
    - Refactoring of `StaticCondensation` code, improved support for
      static condensation with adaptive mesh refinement.
    - Added `matrixsolve` app for testing linear solver behavior with mesh
      files
    - Avoid redundant work when preparing algebraic ghosting in parallel
    - Compatibility with more Abaqus-format mesh files
    - Support for `Elem::opposite_side()` on `Prism` elements
    - Fixes for `BoundaryInfo::sync()` with certain combinations of serial
      and distributed mesh status.  Enabled serialization and/or
      redistribution of distributed boundary meshes with separate interior
      meshes.  Added `MeshBase::interior_mesh()` to track the interior
      mesh of a boundary mesh.
    - Fix unintentionally set `parent()` on 0D side elements of 1D
      interior elements.  Removed deprecated `proxy` option from side and
      edge element construction.
    - Refactoring to use new `Elem::inherit_data_from()` helper.  This
      likely fixes potential issues on certain IGA and p-refined meshes.
    - More exhaustive testing for `BoundaryInfo::sync()` methods,
      including multiple combinations of mesh type, distribution status,
      and dimensionality.
    - Clarifications in Doxygen documentation
    - Minor code simplification/refactoring
    - Update TIMPI submodule
    
      - Enabled `minloc()` and `maxloc()` of arbitrary data types
        overriding `TIMPI::OpFunction<T>`.  Added tests using
        `OpFunction<std::pair>`.
      - Added missing cstdint include, fixing builds with some compilers.
```

Adds the `--fast` option to `scripts/update_and_rebuild_mfem.sh`, which closes https://github.com/idaholab/moose/issues/30731.

Adds `pymongo` to `tools` to support test result database collection, which refs https://github.com/idaholab/moose/issues/30819.

Heinrich-BR committed:
Add --fast option to mfem build script (closes #30731)

roystgnr committed:
Newsletter entries for libMesh update

roystgnr committed:
libMesh submodule update

- Complete overhaul of `VariationalMeshSmoother` code.  This gives
  support for more finite element types and converges to more optimal
  meshes.
- Enabled FE caching as appropriate for variable types with a
  dependency on edge/face orientation.  Added command-line
  `--disable-caching` option to help with debugging.
- Refactoring of `StaticCondensation` code, improved support for
  static condensation with adaptive mesh refinement.
- Added `matrixsolve` app for testing linear solver behavior with mesh
  files
- Avoid redundant work when preparing algebraic ghosting in parallel
- Compatibility with more Abaqus-format mesh files
- Support for `Elem::opposite_side()` on `Prism` elements
- Fixes for `BoundaryInfo::sync()` with certain combinations of serial
  and distributed mesh status.  Enabled serialization and/or
  redistribution of distributed boundary meshes with separate interior
  meshes.  Added `MeshBase::interior_mesh()` to track the interior
  mesh of a boundary mesh.
- Fix unintentionally set `parent()` on 0D side elements of 1D
  interior elements.  Removed deprecated `proxy` option from side and
  edge element construction.
- Refactoring to use new `Elem::inherit_data_from()` helper.  This
  likely fixes potential issues on certain IGA and p-refined meshes.
- More exhaustive testing for `BoundaryInfo::sync()` methods,
  including multiple combinations of mesh type, distribution status,
  and dimensionality.
- Clarifications in Doxygen documentation
- Minor code simplification/refactoring
- Update TIMPI submodule

  - Enabled `minloc()` and `maxloc()` of arbitrary data types
    overriding `TIMPI::OpFunction<T>`.  Added tests using
    `OpFunction<std::pair>`.
  - Added missing cstdint include, fixing builds with some compilers.

Refs #000

moosebuild commented on libmesh:
```diff
@@ -1 +1 @@
-Subproject commit 0846b207f9fd5b70f5f5088684bbfe3d1a7785da
+Subproject commit 307463b8c8aec4b2e7336bf386b5d85610a4fe09
```
**Caution!** This contains a submodule update

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/2959499/), step Generate coverage on 0557d12 wanted to post the following:


## Framework coverage

Coverage did not change


## Modules coverage

Coverage did not change

## Full coverage reports

<details>
<summary>Reports</summary>

- [`framework`](https://mooseframework.inl.gov/docs/PRs/30791/coverage/framework)
- [`chemical_reactions`](https://mooseframework.inl.gov/docs/PRs/30791/coverage/chemical_reactions)
- [`combined`](https://mooseframework.inl.gov/docs/PRs/30791/coverage/combined)
- [`contact`](https://mooseframework.inl.gov/docs/PRs/30791/coverage/contact)
- [`electromagnetics`](https://mooseframework.inl.gov/docs/PRs/30791/coverage/electromagnetics)
- [`external_petsc_solver`](https://mooseframework.inl.gov/docs/PRs/30791/coverage/external_petsc_solver)
- [`fluid_properties`](https://mooseframework.inl.gov/docs/PRs/30791/coverage/fluid_properties)
- [`fsi`](https://mooseframework.inl.gov/docs/PRs/30791/coverage/fsi)
- [`functional_expansion_tools`](https://mooseframework.inl.gov/docs/PRs/30791/coverage/functional_expansion_tools)
- [`geochemistry`](https://mooseframework.inl.gov/docs/PRs/30791/coverage/geochemistry)
- [`heat_transfer`](https://mooseframework.inl.gov/docs/PRs/30791/coverage/heat_transfer)
- [`level_set`](https://mooseframework.inl.gov/docs/PRs/30791/coverage/level_set)
- [`misc`](https://mooseframework.inl.gov/docs/PRs/30791/coverage/misc)
- [`navier_stokes`](https://mooseframework.inl.gov/docs/PRs/30791/coverage/navier_stokes)
- [`optimization`](https://mooseframework.inl.gov/docs/PRs/30791/coverage/optimization)
- [`peridynamics`](https://mooseframework.inl.gov/docs/PRs/30791/coverage/peridynamics)
- [`phase_field`](https://mooseframework.inl.gov/docs/PRs/30791/coverage/phase_field)
- [`porous_flow`](https://mooseframework.inl.gov/docs/PRs/30791/coverage/porous_flow)
- [`ray_tracing`](https://mooseframework.inl.gov/docs/PRs/30791/coverage/ray_tracing)
- [`rdg`](https://mooseframework.inl.gov/docs/PRs/30791/coverage/rdg)
- [`reactor`](https://mooseframework.inl.gov/docs/PRs/30791/coverage/reactor)
- [`richards`](https://mooseframework.inl.gov/docs/PRs/30791/coverage/richards)
- [`scalar_transport`](https://mooseframework.inl.gov/docs/PRs/30791/coverage/scalar_transport)
- [`solid_mechanics`](https://mooseframework.inl.gov/docs/PRs/30791/coverage/solid_mechanics)
- [`solid_properties`](https://mooseframework.inl.gov/docs/PRs/30791/coverage/solid_properties)
- [`stochastic_tools`](https://mooseframework.inl.gov/docs/PRs/30791/coverage/stochastic_tools)
- [`subchannel`](https://mooseframework.inl.gov/docs/PRs/30791/coverage/subchannel)
- [`thermal_hydraulics`](https://mooseframework.inl.gov/docs/PRs/30791/coverage/thermal_hydraulics)
- [`xfem`](https://mooseframework.inl.gov/docs/PRs/30791/coverage/xfem)

</details>



This comment will be updated on new commits.

loganharbour commented:
> So is @loganharbour helping with SAM or should I help with that?

We're having an issue getting to SAM gitlab from within INL at the moment

loganharbour committed:
Add pymongo to moose-tools, bump moose-dev to newer date

This is needed to load data from a mongo database
that stores test results, validation info, and performance
data

refs #30819

roystgnr committed:
Update versioner_hashes.yaml

moosebuild commented on libmesh:
```diff
@@ -1 +1 @@
-Subproject commit 0846b207f9fd5b70f5f5088684bbfe3d1a7785da
+Subproject commit f0a69d16b43414eba6d1c1bfe8482ccd6bda220b
```
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2984838/), step Versioner verify on 3859053 wanted to post the following:

### Versioner templates

Found 14 templates, 0 failed


### Versioner influential files

Found 58 influential files, 4 changed, 0 added, 0 removed

| package   | status   | file                                    |
|-----------|----------|-----------------------------------------|
| libmesh   | CHANGE   | conda/libmesh/meta.yaml                 |
| libmesh   | CHANGE   | libmesh                                 |
| moose-dev | CHANGE   | conda/moose-dev/conda_build_config.yaml |
| moose-dev | CHANGE   | conda/moose-dev/meta.yaml               |

### Versioner versions

Found 9 packages, 2 changed, 0 failed

| package   | status   | hash    | to hash   | version            | to version         |
|-----------|----------|---------|-----------|--------------------|--------------------|
| libmesh   | CHANGE   | 44d69f1 | 985bf51   | 2025.05.23 build 1 | 2025.06.25 build 0 |
| moose-dev | CHANGE   | bb66c5a | c1702b0   | 2025.06.20         | 2025.06.25         |

loganharbour commented:
FYI @milljm - adding `pymongo` to tools for database stuff.

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2987229/), step Versioner verify on 95e7ef5 wanted to post the following:

### Versioner templates

Found 14 templates, 0 failed


### Versioner influential files

Found 58 influential files, 6 changed, 0 added, 0 removed

| package   | status   | file                                    |
|-----------|----------|-----------------------------------------|
| tools     | CHANGE   | conda/tools/meta.yaml                   |
| libmesh   | CHANGE   | conda/libmesh/meta.yaml                 |
| libmesh   | CHANGE   | libmesh                                 |
| moose-dev | CHANGE   | conda/moose-dev/conda_build_config.yaml |
| moose-dev | CHANGE   | conda/moose-dev/meta.yaml               |
| moose-dev | CHANGE   | scripts/update_and_rebuild_mfem.sh      |

### Versioner versions

Found 9 packages, 3 changed, 0 failed

| package   | status   | hash    | to hash   | version            | to version         |
|-----------|----------|---------|-----------|--------------------|--------------------|
| tools     | CHANGE   | 075abd0 | d73012a   | 2025.06.13         | 2025.06.26         |
| libmesh   | CHANGE   | 44d69f1 | 985bf51   | 2025.05.23 build 1 | 2025.06.25 build 0 |
| moose-dev | CHANGE   | bb66c5a | b5c6af5   | 2025.06.20         | 2025.06.26         |

milljm reviewed (approved):
👍

loganharbour commented:
Failures are unrelated and are fixed on next.
